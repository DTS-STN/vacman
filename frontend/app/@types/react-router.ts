import 'react-router';

import type { Namespace } from 'i18next';

declare module 'react-router' {
  interface AppLoadContext {
    nonce: string;
    session: AppSession;
  }

  /**
   * Route handles should export an i18n namespace, if necessary.
   */
  interface RouteHandle {
    i18nNamespace?: Namespace;
  }

  /**
   * A route module exports an optional RouteHandle.
   */
  interface RouteModule {
    handle?: RouteHandle;
  }

  /**
   * Override the default React Router RouteModules
   * to include the new RouteModule type.
   */
  interface RouteModules extends Record<string, RouteModule | undefined> {}
}

export {};
