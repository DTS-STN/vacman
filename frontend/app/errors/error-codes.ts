export type ErrorCode = (typeof ErrorCodes)[keyof typeof ErrorCodes];

export const ErrorCodes = {
  UNCAUGHT_ERROR: 'UNC-0000',

  // auth error codes
  MISCONFIGURED_PROVIDER: 'AUTH-0001',
  ACCESS_FORBIDDEN: 'AUTH-0002',
  MISSING_AUTH_HEADER: 'AUTH-0003',
  DISCOVERY_ENDPOINT_MISSING: 'AUTH-0004',
  MISSING_ID_TOKEN: 'AUTH-0005',

  // component error codes
  MISSING_LANG_PARAM: 'CMP-0001',

  // i18n error codes
  NO_LANGUAGE_FOUND: 'I18N-0001',
  MISSING_TRANSLATION_KEY: 'I18N-0002',

  // instance error codes
  NO_FACTORY_PROVIDED: 'INST-0001',

  // route error codes
  ROUTE_NOT_FOUND: 'RTE-0001',

  // dev-only error codes
  TEST_ERROR_CODE: 'DEV-0001',

  // form error codes
  MISSING_TAB_ID: 'FRM-0000',
  UNRECOGNIZED_ACTION: 'FRM-0001',

  // service error codes
  NO_PROVINCE_FOUND: 'SVC-0001',
  NO_BRANCH_FOUND: 'SVC-0002',
  NO_CLASSIFICATION_FOUND: 'SVC-0003',
  NO_EMPLOYMENT_OPPORTUNITY_TYPE_FOUND: 'SVC-0004',
  NO_EDUCATION_LEVEL_FOUND: 'SVC-0005',
  NO_LANGUAGE_REFERRAL_TYPE_FOUND: 'SVC-0006',
  NO_CITY_FOUND: 'SVC-007',
  NO_WFA_STATUS_FOUND: 'SVC-0008',
  NO_LANGUAGE_OF_CORRESPONDENCE_FOUND: 'SVC-0009',
  NO_DIRECTORATE_FOUND: 'SVC-0010',

  // vacman api error codes
  VACMAN_API_ERROR: 'API-0001',
} as const;
