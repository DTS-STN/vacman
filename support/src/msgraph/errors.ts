import { Data } from 'effect';

/**
 * A custom error type for failures related to the Microsoft Graph API. This
 * class extends `Data.TaggedError` to provide a structured way to represent
 * errors, capturing the original error and an optional descriptive message.
 */
export class MSGraphError extends Data.TaggedError('@support/MSGraphError')<{
  readonly error: unknown;
  readonly message?: string;
}> {}
