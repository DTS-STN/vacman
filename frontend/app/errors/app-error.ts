import type { ErrorCode } from '~/errors/error-codes';
import { ErrorCodes } from '~/errors/error-codes';
import type { HttpStatusCode } from '~/errors/http-status-codes';
import { HttpStatusCodes } from '~/errors/http-status-codes';
import { generateCorrelationId } from '~/utils/correlation';

// Use a dynamic access to avoid bundling server-only modules in the client.
// When running server-side, globalThis.__vacmanGetReqId will be set by the server utils.

type GlobalWithReqId = typeof globalThis & { __vacmanGetReqId?: () => string };

type AppErrorOptions = {
  correlationId?: string;
  httpStatusCode?: HttpStatusCode;
};

/**
 * A generic, top-level error that all application errors should extend.
 * This class *does not* extend Error because React Router will sanitize all Errors when sending them to the client.
 */
export class AppError {
  public readonly name = 'AppError';

  public readonly errorCode: ErrorCode;
  public readonly correlationId: string;
  public readonly stack?: string;
  public readonly httpStatusCode: HttpStatusCode;

  // note: this is intentionally named `msg` instead
  // of `message` to workaround an issue with winston
  // always logging this as the log message when a
  // message is supplied to `log.error(message, error)`
  public readonly msg: string;

  // compatibility getter for standard Error interface
  public get message(): string {
    return this.msg;
  }

  public constructor(msg: string, errorCode: ErrorCode = ErrorCodes.UNCAUGHT_ERROR, opts?: AppErrorOptions) {
    this.errorCode = errorCode;
    this.msg = msg;

    // Prefer explicit correlationId, then request-scoped reqId, then random fallback
    const getReqId = (globalThis as GlobalWithReqId).__vacmanGetReqId;
    const requestReqId: string | undefined = typeof getReqId === 'function' ? getReqId() : undefined;
    this.correlationId = opts?.correlationId ?? requestReqId ?? generateCorrelationId();
    this.httpStatusCode = opts?.httpStatusCode ?? HttpStatusCodes.INTERNAL_SERVER_ERROR;

    try {
      Error.captureStackTrace(this, this.constructor);
    } catch {
      // Stack trace capture is not available in this environment
    }
  }
}

/**
 * Type guard to check if an error is a AppError.
 *
 * Note: this function does not use `instanceof` because the type
 *       information is lost when shipped to the client
 */
export function isAppError(error: unknown): error is AppError {
  return error instanceof Object && 'name' in error && error.name === 'AppError';
}
