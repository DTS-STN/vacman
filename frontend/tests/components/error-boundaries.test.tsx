import { createRoutesStub } from 'react-router';

import { render } from '@testing-library/react';
import { describe, expect, it, vi } from 'vitest';

import {
  BilingualErrorBoundary,
  BilingualNotFound,
  UnilingualErrorBoundary,
  UnilingualNotFound,
} from '~/components/error-boundaries';
import { AppError } from '~/errors/app-error';
import { ErrorCodes } from '~/errors/error-codes';
import { HttpStatusCodes } from '~/errors/http-status-codes';

describe('error-boundaries', () => {
  describe('BilingualErrorBoundary', () => {
    it('should correctly render the bilingual error boundary when it catches a generic error', () => {
      vi.spyOn(console, 'error').mockImplementation(() => {});

      const RoutesStub = createRoutesStub([
        {
          path: '/',
          Component: () => <BilingualErrorBoundary params={{}} error={new Error('Something went wrong')} />,
        },
      ]);

      render(<RoutesStub />);

      expect(document.documentElement).toMatchSnapshot('expected html');
    });

    it('should correctly render the bilingual error boundary when it catches an AppError', () => {
      vi.spyOn(console, 'error').mockImplementation(() => {});

      const RoutesStub = createRoutesStub([
        {
          path: '/',
          Component: () => (
            <BilingualErrorBoundary
              params={{}}
              error={new AppError('Something went wrong', ErrorCodes.UNCAUGHT_ERROR, { correlationId: 'XX-000000' })}
            />
          ),
        },
      ]);

      render(<RoutesStub />);

      expect(document.documentElement).toMatchSnapshot('expected html');
    });
  });

  describe('BilingualNotFound', () => {
    it('should correctly render the bilingual 404 when it catches a 404 error', () => {
      vi.spyOn(console, 'error').mockImplementation(() => {});

      const RoutesStub = createRoutesStub([
        {
          path: '/',
          Component: () => (
            <BilingualNotFound params={{}} error={new Response('Not found', { status: HttpStatusCodes.NOT_FOUND })} />
          ),
        },
      ]);

      render(<RoutesStub />);

      expect(document.documentElement).toMatchSnapshot('expected html');
    });
  });

  describe('UnilingualErrorBoundary', () => {
    it('should correctly render the unilingual error boundary when it catches a generic error', () => {
      vi.spyOn(console, 'error').mockImplementation(() => {});

      const RoutesStub = createRoutesStub([
        {
          path: '/en',
          Component: () => <UnilingualErrorBoundary params={{}} error={new Error('Something went wrong')} />,
        },
      ]);

      render(<RoutesStub initialEntries={['/en']} />);

      expect(document.documentElement).toMatchSnapshot('expected html');
    });

    it('should correctly render the unilingual error boundary when it catches an AppError', () => {
      vi.spyOn(console, 'error').mockImplementation(() => {});

      const RoutesStub = createRoutesStub([
        {
          path: '/en',
          Component: () => (
            <UnilingualErrorBoundary
              params={{}}
              error={new AppError('Something went wrong', ErrorCodes.UNCAUGHT_ERROR, { correlationId: 'XX-000000' })}
            />
          ),
        },
      ]);

      render(<RoutesStub initialEntries={['/en']} />);

      expect(document.documentElement).toMatchSnapshot('expected html');
    });
  });

  describe('UnilingualNotFound', () => {
    it('should correctly render the unilingual 404 when it catches a 404 error', () => {
      vi.spyOn(console, 'error').mockImplementation(() => {});

      const RoutesStub = createRoutesStub([
        {
          path: '/',
          Component: () => (
            <UnilingualNotFound params={{}} error={new Response('Not found', { status: HttpStatusCodes.NOT_FOUND })} />
          ),
        },
      ]);

      render(<RoutesStub />);

      expect(document.documentElement).toMatchSnapshot('expected html');
    });
  });
});
