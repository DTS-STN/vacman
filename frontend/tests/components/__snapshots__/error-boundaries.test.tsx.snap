// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`error-boundaries > BilingualErrorBoundary > should correctly render the bilingual error boundary when it catches a generic error > expected html 1`] = `
<html
  lang="en"
>
  <head>
    <meta
      charset="utf-8"
    />
    <meta
      content="width=device-width, initial-scale=1"
      name="viewport"
    />
    <link
      href="build/stub-path-to-module.js"
      rel="modulepreload"
    />
  </head>
  <body>
    <div>
      <header
        class="border-b-[3px] border-slate-700 print:hidden"
      >
        <div
          id="wb-bnr"
        >
          <div
            class="container flex items-center justify-between gap-6 py-2.5 sm:py-3.5"
          >
            <a
              href="https://canada.ca/"
            >
              <img
                alt="gcweb:header.govt-of-canada.text / gcweb:header.govt-of-canada.text"
                class="h-8 w-auto"
                decoding="async"
                height="28"
                src="https://www.canada.ca/etc/designs/canada/wet-boew/assets/sig-blk-en.svg"
                width="300"
              />
            </a>
          </div>
        </div>
      </header>
      <main
        class="container"
      >
        <div
          class="grid grid-cols-1 gap-6 py-2.5 sm:grid-cols-2 sm:py-3.5"
        >
          <div
            id="english"
            lang="en"
          >
            <div
              class="mt-10 mb-8"
            >
              <h1
                class="font-lato text-3xl font-bold focus-visible:ring-3 after:mt-2 after:block after:h-1.5 after:w-18 after:bg-red-700 my-8"
                id="wb-cont"
                tabindex="-1"
              >
                <span>
                  gcweb:server-error.page-title
                </span>
                <small
                  class="block text-2xl font-normal text-neutral-500"
                >
                  {"key":"gcweb:server-error.page-subtitle","options":{"statusCode":500}}
                </small>
              </h1>
            </div>
            <p
              class="mb-8 text-lg text-gray-500"
            >
              gcweb:server-error.page-message
            </p>
          </div>
          <div
            id="french"
            lang="fr"
          >
            <div
              class="mt-10 mb-8"
            >
              <h1
                class="font-lato text-3xl font-bold focus-visible:ring-3 after:mt-2 after:block after:h-1.5 after:w-18 after:bg-red-700 my-8"
                id="wb-cont"
                tabindex="-1"
              >
                <span>
                  gcweb:server-error.page-title
                </span>
                <small
                  class="block text-2xl font-normal text-neutral-500"
                >
                  {"key":"gcweb:server-error.page-subtitle","options":{"statusCode":500}}
                </small>
              </h1>
            </div>
            <p
              class="mb-8 text-lg text-gray-500"
            >
              gcweb:server-error.page-message
            </p>
          </div>
        </div>
      </main>
      <footer
        class="mt-8 bg-stone-50 print:hidden"
        id="wb-info"
        tabindex="-1"
      >
        <div
          class="container flex items-center justify-end gap-6 py-2.5 sm:py-3.5"
        >
          <div>
            <h2
              class="sr-only"
            >
              <span
                lang="en"
              >
                gcweb:footer.about-site
              </span>
               / 
              <span
                lang="fr"
              >
                gcweb:footer.about-site
              </span>
            </h2>
            <div>
              <img
                alt="gcweb:footer.gc-symbol / gcweb:footer.gc-symbol"
                class="h-10 w-auto"
                height="71"
                src="https://www.canada.ca/etc/designs/canada/wet-boew/assets/wmms-blk.svg"
                width="300"
              />
            </div>
          </div>
        </div>
      </footer>
      <link
        rel="modulepreload"
      />
      <link
        rel="modulepreload"
      />
      <script>
         
      </script>
      <script
        async=""
        type="module"
      >
         
      </script>
      <script
        src="/api/client-env?v=undefined"
      />
    </div>
  </body>
</html>
`;

exports[`error-boundaries > BilingualErrorBoundary > should correctly render the bilingual error boundary when it catches an AppError > expected html 1`] = `
<html
  lang="en"
>
  <head>
    <meta
      charset="utf-8"
    />
    <meta
      content="width=device-width, initial-scale=1"
      name="viewport"
    />
  </head>
  <body>
    <div>
      <header
        class="border-b-[3px] border-slate-700 print:hidden"
      >
        <div
          id="wb-bnr"
        >
          <div
            class="container flex items-center justify-between gap-6 py-2.5 sm:py-3.5"
          >
            <a
              href="https://canada.ca/"
            >
              <img
                alt="gcweb:header.govt-of-canada.text / gcweb:header.govt-of-canada.text"
                class="h-8 w-auto"
                decoding="async"
                height="28"
                src="https://www.canada.ca/etc/designs/canada/wet-boew/assets/sig-blk-en.svg"
                width="300"
              />
            </a>
          </div>
        </div>
      </header>
      <main
        class="container"
      >
        <div
          class="grid grid-cols-1 gap-6 py-2.5 sm:grid-cols-2 sm:py-3.5"
        >
          <div
            id="english"
            lang="en"
          >
            <div
              class="mt-10 mb-8"
            >
              <h1
                class="font-lato text-3xl font-bold focus-visible:ring-3 after:mt-2 after:block after:h-1.5 after:w-18 after:bg-red-700 my-8"
                id="wb-cont"
                tabindex="-1"
              >
                <span>
                  gcweb:server-error.page-title
                </span>
                <small
                  class="block text-2xl font-normal text-neutral-500"
                >
                  {"key":"gcweb:server-error.page-subtitle","options":{"statusCode":500}}
                </small>
              </h1>
            </div>
            <p
              class="mb-8 text-lg text-gray-500"
            >
              gcweb:server-error.page-message
            </p>
            <ul
              class="list-disc space-y-1 pl-7 text-gray-800"
            >
              <li>
                {"key":"gcweb:server-error.error-code","options":{"values":{"errorCode":"UNC-0000"}}}
              </li>
              <li>
                {"key":"gcweb:server-error.correlation-id","options":{"values":{"correlationId":"XX-000000"}}}
              </li>
            </ul>
          </div>
          <div
            id="french"
            lang="fr"
          >
            <div
              class="mt-10 mb-8"
            >
              <h1
                class="font-lato text-3xl font-bold focus-visible:ring-3 after:mt-2 after:block after:h-1.5 after:w-18 after:bg-red-700 my-8"
                id="wb-cont"
                tabindex="-1"
              >
                <span>
                  gcweb:server-error.page-title
                </span>
                <small
                  class="block text-2xl font-normal text-neutral-500"
                >
                  {"key":"gcweb:server-error.page-subtitle","options":{"statusCode":500}}
                </small>
              </h1>
            </div>
            <p
              class="mb-8 text-lg text-gray-500"
            >
              gcweb:server-error.page-message
            </p>
            <ul
              class="list-disc space-y-1 pl-7 text-gray-800"
            >
              <li>
                {"key":"gcweb:server-error.error-code","options":{"values":{"errorCode":"UNC-0000"}}}
              </li>
              <li>
                {"key":"gcweb:server-error.correlation-id","options":{"values":{"correlationId":"XX-000000"}}}
              </li>
            </ul>
          </div>
        </div>
      </main>
      <footer
        class="mt-8 bg-stone-50 print:hidden"
        id="wb-info"
        tabindex="-1"
      >
        <div
          class="container flex items-center justify-end gap-6 py-2.5 sm:py-3.5"
        >
          <div>
            <h2
              class="sr-only"
            >
              <span
                lang="en"
              >
                gcweb:footer.about-site
              </span>
               / 
              <span
                lang="fr"
              >
                gcweb:footer.about-site
              </span>
            </h2>
            <div>
              <img
                alt="gcweb:footer.gc-symbol / gcweb:footer.gc-symbol"
                class="h-10 w-auto"
                height="71"
                src="https://www.canada.ca/etc/designs/canada/wet-boew/assets/wmms-blk.svg"
                width="300"
              />
            </div>
          </div>
        </div>
      </footer>
      <script
        src="/api/client-env?v=undefined"
      />
    </div>
  </body>
</html>
`;

exports[`error-boundaries > BilingualNotFound > should correctly render the bilingual 404 when it catches a 404 error > expected html 1`] = `
<html
  lang="en"
>
  <head>
    <meta
      charset="utf-8"
    />
    <meta
      content="width=device-width, initial-scale=1"
      name="viewport"
    />
  </head>
  <body>
    <div>
      <header
        class="border-b-[3px] border-slate-700 print:hidden"
      >
        <div
          id="wb-bnr"
        >
          <div
            class="container flex items-center justify-between gap-6 py-2.5 sm:py-3.5"
          >
            <a
              href="https://canada.ca/"
            >
              <img
                alt="gcweb:header.govt-of-canada.text"
                class="h-8 w-auto"
                decoding="async"
                height="28"
                src="https://www.canada.ca/etc/designs/canada/wet-boew/assets/sig-blk-en.svg"
                width="300"
              />
              <span
                class="sr-only"
              >
                / 
                <span
                  lang="fr"
                >
                  gcweb:header.govt-of-canada.text
                </span>
              </span>
            </a>
          </div>
        </div>
      </header>
      <main
        class="container"
      >
        <div
          class="grid grid-cols-1 gap-6 py-2.5 sm:grid-cols-2 sm:py-3.5"
        >
          <div
            id="english"
            lang="en"
          >
            <div
              class="mt-10 mb-8"
            >
              <h1
                class="font-lato text-3xl font-bold focus-visible:ring-3 after:mt-2 after:block after:h-1.5 after:w-18 after:bg-red-700 my-8"
                id="wb-cont"
                tabindex="-1"
              >
                <span>
                  gcweb:not-found.page-title
                </span>
                <small
                  class="block text-2xl font-normal text-neutral-500"
                >
                  gcweb:not-found.page-subtitle
                </small>
              </h1>
            </div>
            <p
              class="mb-8 text-lg text-gray-500"
            >
              gcweb:not-found.page-message
            </p>
          </div>
          <div
            id="french"
            lang="fr"
          >
            <div
              class="mt-10 mb-8"
            >
              <h1
                class="font-lato text-3xl font-bold focus-visible:ring-3 after:mt-2 after:block after:h-1.5 after:w-18 after:bg-red-700 my-8"
                id="wb-cont"
                tabindex="-1"
              >
                <span>
                  gcweb:not-found.page-title
                </span>
                <small
                  class="block text-2xl font-normal text-neutral-500"
                >
                  gcweb:not-found.page-subtitle
                </small>
              </h1>
            </div>
            <p
              class="mb-8 text-lg text-gray-500"
            >
              gcweb:not-found.page-message
            </p>
          </div>
        </div>
      </main>
      <footer
        class="bg-stone-50 print:hidden"
        id="wb-info"
        tabindex="-1"
      >
        <div
          class="container flex items-center justify-end gap-6 py-2.5 sm:py-3.5"
        >
          <div>
            <h2
              class="sr-only"
            >
              <span
                lang="en"
              >
                gcweb:footer.about-site
              </span>
               / 
              <span
                lang="fr"
              >
                gcweb:footer.about-site
              </span>
            </h2>
            <div>
              <img
                alt="gcweb:footer.gc-symbol / gcweb:footer.gc-symbol"
                class="h-10 w-auto"
                height="71"
                src="https://www.canada.ca/etc/designs/canada/wet-boew/assets/wmms-blk.svg"
                width="300"
              />
            </div>
          </div>
        </div>
      </footer>
      <script
        src="/api/client-env?v=undefined"
      />
    </div>
  </body>
</html>
`;

exports[`error-boundaries > UnilingualErrorBoundary > should correctly render the unilingual error boundary when it catches a generic error > expected html 1`] = `
<html
  lang="en"
>
  <head>
    <meta
      charset="utf-8"
    />
    <meta
      content="width=device-width, initial-scale=1"
      name="viewport"
    />
  </head>
  <body>
    <div>
      <header
        class="border-b-[3px] border-slate-700 print:hidden"
      >
        <div
          id="wb-bnr"
        >
          <div
            class="container flex items-center justify-between gap-6 py-2.5 sm:py-3.5"
          >
            <a
              href="https://canada.ca/"
            >
              <img
                alt="gcweb:header.govt-of-canada.text"
                class="h-8 w-auto"
                decoding="async"
                height="28"
                src="https://www.canada.ca/etc/designs/canada/wet-boew/assets/sig-blk-en.svg"
                width="300"
              />
            </a>
          </div>
        </div>
      </header>
      <main
        class="container"
      >
        <div
          class="mt-10 mb-8"
        >
          <h1
            class="font-lato text-3xl font-bold focus-visible:ring-3 after:mt-2 after:block after:h-1.5 after:w-18 after:bg-red-700 my-8"
            id="wb-cont"
            tabindex="-1"
          >
            <span>
              gcweb:server-error.page-title
            </span>
            <small
              class="block text-2xl font-normal text-neutral-500"
            >
              {"key":"gcweb:server-error.page-subtitle","options":{"statusCode":500}}
            </small>
          </h1>
        </div>
        <p
          class="mb-8 text-lg text-gray-500"
        >
          gcweb:server-error.page-message
        </p>
      </main>
      <footer
        class="mt-8 bg-stone-50 print:hidden"
        id="wb-info"
        tabindex="-1"
      >
        <div
          class="container flex items-center justify-end gap-6 py-2.5 sm:py-3.5"
        >
          <div>
            <h2
              class="sr-only"
            >
              gcweb:footer.about-site
            </h2>
            <div>
              <img
                alt="gcweb:footer.gc-symbol"
                class="h-10 w-auto"
                height="71"
                src="https://www.canada.ca/etc/designs/canada/wet-boew/assets/wmms-blk.svg"
                width="300"
              />
            </div>
          </div>
        </div>
      </footer>
      <script
        src="/api/client-env?v=undefined"
      />
    </div>
  </body>
</html>
`;

exports[`error-boundaries > UnilingualErrorBoundary > should correctly render the unilingual error boundary when it catches an AppError > expected html 1`] = `
<html
  lang="en"
>
  <head>
    <meta
      charset="utf-8"
    />
    <meta
      content="width=device-width, initial-scale=1"
      name="viewport"
    />
  </head>
  <body>
    <div>
      <header
        class="border-b-[3px] border-slate-700 print:hidden"
      >
        <div
          id="wb-bnr"
        >
          <div
            class="container flex items-center justify-between gap-6 py-2.5 sm:py-3.5"
          >
            <a
              href="https://canada.ca/"
            >
              <img
                alt="gcweb:header.govt-of-canada.text"
                class="h-8 w-auto"
                decoding="async"
                height="28"
                src="https://www.canada.ca/etc/designs/canada/wet-boew/assets/sig-blk-en.svg"
                width="300"
              />
            </a>
          </div>
        </div>
      </header>
      <main
        class="container"
      >
        <div
          class="mt-10 mb-8"
        >
          <h1
            class="font-lato text-3xl font-bold focus-visible:ring-3 after:mt-2 after:block after:h-1.5 after:w-18 after:bg-red-700 my-8"
            id="wb-cont"
            tabindex="-1"
          >
            <span>
              gcweb:server-error.page-title
            </span>
            <small
              class="block text-2xl font-normal text-neutral-500"
            >
              {"key":"gcweb:server-error.page-subtitle","options":{"statusCode":500}}
            </small>
          </h1>
        </div>
        <p
          class="mb-8 text-lg text-gray-500"
        >
          gcweb:server-error.page-message
        </p>
        <ul
          class="list-disc space-y-1 pl-7 text-gray-800"
        >
          <li>
            {"key":"gcweb:server-error.error-code","options":{"values":{"errorCode":"UNC-0000"}}}
          </li>
          <li>
            {"key":"gcweb:server-error.correlation-id","options":{"values":{"correlationId":"XX-000000"}}}
          </li>
        </ul>
      </main>
      <footer
        class="mt-8 bg-stone-50 print:hidden"
        id="wb-info"
        tabindex="-1"
      >
        <div
          class="container flex items-center justify-end gap-6 py-2.5 sm:py-3.5"
        >
          <div>
            <h2
              class="sr-only"
            >
              gcweb:footer.about-site
            </h2>
            <div>
              <img
                alt="gcweb:footer.gc-symbol"
                class="h-10 w-auto"
                height="71"
                src="https://www.canada.ca/etc/designs/canada/wet-boew/assets/wmms-blk.svg"
                width="300"
              />
            </div>
          </div>
        </div>
      </footer>
      <script
        src="/api/client-env?v=undefined"
      />
    </div>
  </body>
</html>
`;

exports[`error-boundaries > UnilingualNotFound > should correctly render the unilingual 404 when it catches a 404 error > expected html 1`] = `
<html>
  <head>
    <meta
      charset="utf-8"
    />
    <meta
      content="width=device-width, initial-scale=1"
      name="viewport"
    />
  </head>
  <body>
    <div>
      <header
        class="border-b-[3px] border-slate-700 print:hidden"
      >
        <div
          id="wb-bnr"
        >
          <div
            class="container flex items-center justify-between gap-6 py-2.5 sm:py-3.5"
          >
            <a
              href="https://canada.ca/"
            >
              <img
                alt="gcweb:header.govt-of-canada.text"
                class="h-8 w-auto"
                decoding="async"
                height="28"
                src="https://www.canada.ca/etc/designs/canada/wet-boew/assets/sig-blk-en.svg"
                width="300"
              />
            </a>
          </div>
        </div>
      </header>
      <main
        class="container"
      >
        <div
          class="mt-10 mb-8"
        >
          <h1
            class="font-lato text-3xl font-bold focus-visible:ring-3 after:mt-2 after:block after:h-1.5 after:w-18 after:bg-red-700 my-8"
            id="wb-cont"
            tabindex="-1"
          >
            <span>
              gcweb:not-found.page-title
            </span>
            <small
              class="block text-2xl font-normal text-neutral-500"
            >
              gcweb:not-found.page-subtitle
            </small>
          </h1>
        </div>
        <p
          class="mb-8 text-lg text-gray-500"
        >
          gcweb:not-found.page-message
        </p>
      </main>
      <footer
        class="bg-stone-50 print:hidden"
        id="wb-info"
        tabindex="-1"
      >
        <div
          class="container flex items-center justify-end gap-6 py-2.5 sm:py-3.5"
        >
          <div>
            <h2
              class="sr-only"
            >
              gcweb:footer.about-site
            </h2>
            <div>
              <img
                alt="gcweb:footer.gc-symbol"
                class="h-10 w-auto"
                height="71"
                src="https://www.canada.ca/etc/designs/canada/wet-boew/assets/wmms-blk.svg"
                width="300"
              />
            </div>
          </div>
        </div>
      </footer>
      <script
        src="/api/client-env?v=undefined"
      />
    </div>
  </body>
</html>
`;
